edflow/data/util/util_dsets.py

```diff
@@ -139,12 +137,13 @@ def get_example(self, i):
             join_indices = self.prng.choice(choices, self.n_joins - 1, replace=replace)
         join_indices = np.concatenate([[i], join_indices])
 
-        examples = dict()
+        examples = list()
+        indices = list()
         for i_join, idx in enumerate(join_indices):
```

Comment:
No need to iterate, `DatasetMixin` can work with lists and arrays.

Reply:
done!